api/rosetta/data_transaction_integration_test.go

```diff
@@ -0,0 +1,554 @@
+		},
+		{
+			name: "missing blockchain name",
+			request: rosetta.TransactionRequest{
+				NetworkID: identifier.Network{
+					Blockchain: "",
+					Network:    dps.FlowTestnet.String(),
+				},
+				BlockID:       testBlock,
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "blockchain identifier: blockchain field is empty",
+		},
+		{
+			name: "invalid blockchain name",
+			request: rosetta.TransactionRequest{
+				NetworkID: identifier.Network{
+					Blockchain: invalidBlockchain,
+					Network:    dps.FlowTestnet.String(),
+				},
+				BlockID:       testBlock,
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidNetwork,
+			wantRosettaErrorDescription: fmt.Sprintf("invalid network identifier blockchain (have: %s, want: %s)", invalidBlockchain, dps.FlowBlockchain),
+			wantRosettaErrorDetails:     map[string]interface{}{"blockchain": invalidBlockchain, "network": dps.FlowTestnet.String()},
+		},
+		{
+			name: "missing network name",
+			request: rosetta.TransactionRequest{
+				NetworkID: identifier.Network{
+					Blockchain: dps.FlowBlockchain,
+					Network:    "",
+				},
+				BlockID:       testBlock,
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "blockchain identifier: network field is empty",
+		},
+		{
+			name: "invalid network name",
+			request: rosetta.TransactionRequest{
+				NetworkID: identifier.Network{
+					Blockchain: dps.FlowBlockchain,
+					Network:    invalidNetwork,
+				},
+				BlockID:       testBlock,
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidNetwork,
+			wantRosettaErrorDescription: fmt.Sprintf("invalid network identifier network (have: %s, want: %s)", invalidNetwork, dps.FlowTestnet.String()),
+			wantRosettaErrorDetails:     map[string]interface{}{"blockchain": dps.FlowBlockchain, "network": invalidNetwork},
+		},
+		{
+			name: "missing block height and hash",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID: identifier.Block{
+					Index: 0,
+					Hash:  "",
+				},
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "block identifier: at least one of hash or index is required",
+		},
+		{
+			name: "invalid length of block id",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID: identifier.Block{
+					Index: 0,
+					Hash:  trimmedBlockHash,
+				},
+				TransactionID: testTx,
+			},
```

Comment:
```suggestion
			name: "invalid length of block id",
			request: rosetta.TransactionRequest{
				NetworkID: defaultNetwork(),
				BlockID: identifier.Block{
					Hash:  trimmedBlockHash,
				},
				TransactionID: testTx,
			},
```

AFAIK the `Index` attribute doesn't need to be specified here

Reply:
Indeed it doesn't, I wanted to make it clear that the `Index` is initialized but it's the other value (`Hash`) that triggers the error. I'll change it to a non-zero `Index`